clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockOutputStream.java

```diff
@@ -259,10 +259,10 @@ private void ensureNotFinished() {
     @Override
     public void close() throws IOException {
         if (!finished) {
+            // basicallly flush the buffer writing the last block
             writeBlock();
+            // write the end block and finish the stream
             writeEndMark();
-            flush();
```

Comment:
Regarding your latter point, I think you are conflating two different things. This `close` method does the same as `flush` in that it calls `writeBlock`. So it is correctly implemented. The thing that changed is that for the underlying stream, it calls `close` without calling `flush`.

Anyway, we can be conservative and do a `out.flush()` before `out.close()`. I think that's clearer than calling the `flush` since `finished = true` already.